test/e2e/tools/event-writer/install-ebpf-xdp.ps1

```diff
@@ -0,0 +1,754 @@
+      [String] $LocalPath = "$env:TEMP"
+   )
+
+   [Boolean] $isSuccess = $true
+
+   Try
+   {
+      Write-Host 'Installing extended Berkley Packet Filter for Windows'
+      If(-Not (Assert-TestSigningIsEnabled))
+      {
+         If(-Not (Enable-TestSigning -Reboot)) { Throw }
+      }
+
+      If(Assert-SoftwareInstalled -ServiceName:"eBPFCore")
+      {
+         Write-Host 'extended Berkley Packet Filter for Windows is already installed'
+         return $isSuccess
+      }
+
+      Write-Host 'Installing extended Berkley Packet Filter for Windows'
+      # Download eBPF-for-Windows.
+      $packageEbpfUrl  = "https://github.com/microsoft/ebpf-for-windows/releases/download/Release-v0.21.1/ebpf-for-windows.x64.0.21.1.msi"
+      Invoke-WebRequest -Uri $packageEbpfUrl -OutFile "$LocalPath\ebpf-for-windows.x64.0.21.1.msi"
+
+      Start-Process -FilePath "$($env:WinDir)\System32\MSIExec.exe" -ArgumentList @("/i", "$LocalPath\ebpf-for-windows.x64.0.21.1.msi", "/qn", "INSTALLFOLDER=`"$($env:ProgramFiles)\ebpf-for-windows`"", "ADDLOCAL=eBPF_Runtime_Components") -PassThru | Wait-Process
+      If(-Not (Assert-SoftwareInstalled -ServiceName:'eBPFCore' -Silent) -Or
+         -Not (Assert-SoftwareInstalled -ServiceName:'NetEbpfExt' -Silent))
+      {
+         Write-Error -Message:"`eBPF service failed to install"
+         Throw
+      }
+
+      $isSuccess = Assert-SoftwareInstalled -ServiceName:"eBPFCore"
+   }
+   Catch
+   {
+      $isSuccess = $false
+      Write-Host "EBPF install failed : $_"
+      Uninstall-eBPF
+   }
+
+   Return $isSuccess
+}
+
+<#
+ .Name
+   Install-XDP
+
+ .Synopsis
+   Installs the eXpress Data Path for Windows service.
+
+ .Description
+   Returns TRUE if the eXpress Data Path for Windows service is installed successfully, otherwise FALSE.
+
+ .Parameter LocalPath
+   Local directory to the eXpress Data Path for Windows service binaries.
+   Default location is $env:LocalAppData\Temp
+
+ .Example
+   # Install the eXpress Data Path service
+   Install-XDP -LocalPath:"$env:TEMP"
+#>
+Function Install-XDP
+{
+   [cmdletbinding(DefaultParameterSetName='Default')]
+
+   Param
+   (
+      [Parameter(ParameterSetName='Default',Mandatory=$false)]
+      [ValidateScript({Test-Path $_ -PathType:'Container'})]
+      [String] $LocalPath = "$env:TEMP"
+   )
+
+   [Boolean] $isSuccess = $true
+
+   Try
+   {
+      If(Assert-SoftwareInstalled -SoftwareName:'XDP for Windows' -Silent)
+      {
+         Write-Host 'XDP for Windows is already installed'
+         return $isSuccess
+      }
+
+      # Download XDP-for-Windows.
+      Write-Host 'Installing eXpress Data Path for Windows'
+      CertUtil.exe -addstore Root "$LocalPath\xdp.cer"
+      CertUtil.exe -addstore TrustedPublisher "$LocalPath\xdp.cer"
+      Invoke-WebRequest -Uri "https://github.com/microsoft/xdp-for-windows/releases/download/v1.2.0-prerelease-793dc2a0/xdp-for-windows.x64.1.2.0-prerelease-793dc2a0.msi" -OutFile "$LocalPath\xdp-for-windows.msi"
```

Comment:
In the `event_writer` project, XDP version 1.1.0 was installed. Do you want to sync these versions?

Reply:
The SDK version on nuget.org is still 1.1.0 which has not been updated but for run time we need the 1.2..0-prerelease.